types/block.go

```diff
@@ -741,6 +740,160 @@ func (cs *CommitSig) FromProto(csp tmproto.CommitSig) error {
 
 //-------------------------------------
 
+// ExtendedCommitSig is similar to CommitSig, except it also contains the vote
+// extension and vote extension signature related to this commit.
+type ExtendedCommitSig struct {
+	BlockIDFlag        BlockIDFlag `json:"block_id_flag"`
+	ValidatorAddress   Address     `json:"validator_address"`
+	Timestamp          time.Time   `json:"timestamp"`
+	Signature          []byte      `json:"signature"`
+	VoteExtension      []byte      `json:"extension"`
+	ExtensionSignature []byte      `json:"extension_signature"`
```

Comment:
What is the JSON encoding for, do we actually render this as JSON somewhere?

Reply:
I wondered the same thing, but noted we also have tags for the existing Commit. I'm not sure if that's for RPC, legacy storage, or inertia. It's harmless to have the tags anyway, and we could do a cleanup pass if they turn out not to be needed (I haven't figured out the answer yet).

Reply:
Oh, I meant on the type more generally. I don't believe we will actually expose this type via RPC?

Reply:
I wasn't sure. You can query blocks and headers there, so I thought we might.

Reply:
That's true, but this type does not appear in either. It's an intermediary type for shuttle values into the DB and passing them to peers. It only every materializes to outside consumers (p2p and the db) in its proto message counterpart of the same name.